toolchain/check/testdata/interop/cpp/function_decl.carbon

```diff
@@ -45,9 +45,55 @@ fn F() {
   Cpp.foo_float();
 }
 
+// --- signed_int_param_function_decl.h
+
+auto foo(signed int a) -> void;
+
+// --- import_signed_int_param_function_decl.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "signed_int_param_function_decl.h";
+
+fn F() {
+  Cpp.foo(1);
```

Comment:
Should we test the boundary conditions too for these tests? For this case, we could test passing INT_MAX and INT_MIN, and a fail test that passes values outside that range?